mail_show_follower/models/res_partner.py

```diff
@@ -0,0 +1,30 @@
+from odoo import api, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    @api.model
+    def _get_excluded_partners_domain(self):
+        return [
+            ("active", "in", (True, False)),
+            ("show_in_cc", "=", False),
+            ("groups_id", "not in", self.env.ref("base.group_portal").ids),
+        ]
+
+    @api.model
+    def _get_excluded_partners(self):
+        domain = self._get_excluded_partners_domain()
+        return self.env["res.users"].search(domain).mapped("partner_id")
+
+    def _show_in_cc(self, show_internal_users):
+        self.ensure_one()
+        if not self.user_ids:
+            return True
+        excluded_partners = self._get_excluded_partners()
+        if show_internal_users:
+            return self not in excluded_partners and self.user_ids.mapped("show_in_cc")
+        else:
+            return self not in excluded_partners and self.env.ref(
+                "base.group_portal"
+            ) not in self.user_ids.mapped("groups_id")
```

Comment:
issue(performance): this implementation is not a good idea.

Main method `_show_in_cc()` is called in loop from `MailMail._send()`. For each loop iteration, it will make a search. Making a search involves:
1. Flushing changes to Postgres.
2. Clearing some caches.
3. Asking Postgres for some more data.

We should avoid interaction with DB as much as possible to avoid performance penalties. If we have to contact it, then we should batch it as much as possible.

Here you have an alternative implementation that reuses ORM cache and will be much faster when called in loop. I take the chance to add some comments that will make it more clear what the method does:

```suggestion
    def _filter_shown_in_cc(self, show_internal_users):
        """Get partners that should be displayed as CC on mails."""
        # Never display hidden users
        result = self.filtered_domain(
            [
                "|",
                ("user_ids", "=", False),
                ("user_ids.show_in_cc", "=", True),
            ]
        )
        # Remove internal users from result if needed
        if not show_internal_users:
            internal_users = result.filtered_domain(
                [
                    ("user_ids.active", "=", True),
                    ("user_ids.groups_id", "in",  self.env.ref("base.group_user").ids),
                ]
            )
            result -= internal_users
        return result
```

By using `filtered_domain`, you don't have to ask the ORM again, and you batch operations.